app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+    private val assistanceData = registrationData.assistanceData!!
+    private val reviewData = registrationData.reviewData!!
+
+    /**
+     * Transforms the [registrationData] and [partnerInformation] into a [RockyRequest]
+     *
+     * @throws [InvalidRegistrationException] if unable to validate [RegistrationData]
+     */
+    fun transform(
+        unknown: UnknownDataSource
+    ): RockyRequest {
+
+        val body = RockyRequestBody(
+            lang = reviewData.formLanguage,
+            phoneType = null, // TODO I can't find where/what is supposed to populate this
+            partnerId = partnerInformation.partnerId,
+            optInEmail = additionalInfoData.hasOptedIntoNewsUpdates,
+            optInSms = additionalInfoData.hasOptedIntoNewsCallAndText,
+            optInVolunteer = additionalInfoData.hasOptedForVolunteerText,
+            // TODO All of these data sources are unknown, I think they may be from partner information, but don't know for sure
+            partnerOptInSms = unknown.partnerOptInSms,
+            partnerOptInEmail = unknown.partnerOptInEmail,
+            partnerOptInVolunteer = unknown.partnerOptInVolunteer,
+            finishWithState = true, // TODO is this a default value?
+            createdViaApi = true, // TODO is this a default value?
+            sourceTrackingId = unknown.sourceTrackingId,
+            partnerTrackingId = unknown.partnerTrackingId,
+            geoLocation = unknown.geoLocation,
+            openTrackingId = unknown.openTrackingId,
+            voterRecordsRequest = buildVoterRecordsRequest()
+        )
+
+        return RockyRequest(body)
+    }
+
+    private fun buildVoterRecordsRequest(): VoterRecordsRequest {
+        return VoterRecordsRequest(
+            type = "registration", // TODO is this a default value?
+            // TODO is this from somewhere else, I can't find a source?
+            // TODO Does this _have_ to be a full datetime stamp? It is in the example, but I don't see a formatter anywhere in the project
+            //  Looks like not? From old RockyRequest.java: Date generatedDate = Dates.parseISO8601_Date(Db.getString(cursor, GENERATED_DATE));
+            generatedDate = Dates.formatAsISO8601_Date(Date()),
+            canvasserName = partnerInformation.canvasserName,
+            voterRegistration = buildVoterRegistration()
+        )
+    }
+
+    private fun buildVoterRegistration(): VoterRegistration {
+        val mailingAddress = if (addressData.isMailingAddressDifferent) {
+            addressData.mailingAddress ?: throw InvalidRegistrationException(
+                "addressData.isMailingAddressDifferent is null",
+                R.string.different_mailing_address_not_filled_out
+            )
+        } else {
+            null
+        }
+
+        // TODO Is this the way that we should be specifying race? It's required in the API, but not in the app
+        val race = (additionalInfoData.race ?: Race.DECLINE).toString()
```

Comment:
A response is required, "decline" counts.

The `decline` option is the default in the spinner, so there should always be a value, having a default here as well doesn't hurt.